File: library/src/main/java/com/devbrackets/android/recyclerext/adapter/delegate/DelegateApi.java
package com.devbrackets.android.recyclerext.adapter.delegate;

public interface DelegateApi<T> {

    /**
     * Retrieves the item associated with the <code>position</code>
     *
     * @param position The position to get the item for
     * @return The item in the <code>position</code>
     */
    T getItem(int position);

    int getItemViewType(int position);
}

File: library/src/main/kotlin/com/devbrackets/android/recyclerext/adapter/delegate/DelegateApi.kt
package com.devbrackets.android.recyclerext.adapter.delegate

interface DelegateApi<T> {
  /**
   * Retrieves the item associated with the `position`
   *
   * @param position The position to get the item for
   * @return The item in the `position`
   */
  fun getItem(position: Int): T
  fun getItemViewType(adapterPosition: Int): Int
}